lib/engine/config/headways.ex

```diff
@@ -31,20 +31,16 @@ defmodule Engine.Config.Headways do
 
   @spec parse(map()) :: [Headway.t()]
   def parse(map) do
-    Enum.flat_map(map, fn {group, group_conf} ->
-      Enum.flat_map(group_conf, fn {time_period, headway_conf} ->
-        case Headway.from_map(group, time_period, headway_conf) do
-          {:ok, conf} ->
-            [conf]
-
-          :error ->
-            Logger.error(
-              "event=group_headways_parse_error headway_id=#{inspect({group, time_period})} group_conf=#{inspect(group_conf)}"
-            )
-
-            []
-        end
-      end)
-    end)
+    for {group, periods} <- map,
+        {period, %{"range_low" => low, "range_high" => high}} <- periods,
+        time_period = parse_time_period(period) do
```

Comment:
since `time_period` could be `nil` and then is passed to the `Headway` struct, shouldn't the typespec for `Headway.time_period()` also include `| nil`?

Reply:
We're relying on the fact that falsy values act like a filter, so they actually get skipped and we don't create `Headway` structs for those.